crypto/Pos/Crypto/HD.hs

```diff
@@ -125,7 +125,7 @@ packHDAddressAttr (HDPassphrase passphrase) path = do
         CryptoPassed p  -> HDAddressPayload p
 
 -- | Try to decrypt HDAddressPayload using HDPassphrase.
-unpackHDAddressAttr :: MonadFail m => HDPassphrase -> HDAddressPayload -> m [Word32]
+unpackHDAddressAttr :: HDPassphrase -> HDAddressPayload -> Maybe [Word32]
```

Comment:
Why not `Either`? We are losing some information here. I think it's better for the caller to decide whether that information is useful.

Reply:
When I was instantiating these types, I was studying all call sites. For this function, its type is always instantiated to `Maybe`, and that is how I rewrote it.